### PR TITLE
Enforce constraints

### PR DESCRIPTION
**Pull request by jeremykross**
@nicktate @normanjoyner

### CODE REVIEW

File: lib/kubernetes-api.js [hunk 1]
```diff
@@ -137,6 +137,16 @@ class KubernetesApi extends ApiImplementation {
         });
     }
 
+    enforceConstraints(app) {
+        this.getHosts((err, hosts) => {
+            if(err) {
+                return console.error('failed to get hosts in enforceConstraints');
+            }
+
+            this.enforceConstraints(hosts, app);
```
Comment (nicktate) [inline lib/kubernetes-api.js]: Javascript does not support function overloading. This will not work as intended.

File: lib/kubernetes-api.js [hunk 2]
```diff
@@ -383,14 +393,18 @@ class KubernetesApi extends ApiImplementation {
...
-            return this.updateApplication(appId, {count: newCount}, cb);
+            return this.updateApplication(appId, {count: newCount}, cb, false);
         });
     }
 
-    updateApplication(appId, csAppDesc, cb) {
-        const{ handler, errorHandler } = this.sanitizeCallback(cb);
+    updateApplication(appId, csAppDesc, cb, shouldEnforceConstraints) {
+
+        shouldEnforceConstraints = shouldEnforceConstraints || true;
```
Comment (nicktate) [inline lib/kubernetes-api.js]: this will always evaluate as true

Comment (jeremykross) [inline lib/kubernetes-api.js]: :man_facepalming: - ha, thanks.

**Comment by nicktate** [comment]
can you sync with latest. we removed `scaleUp / scaleDown / _scale` from the `k8s.bridge` and put it in the `abstractions.api`

**Comment by nicktate** [comment]
https://github.com/containership/containership.k8s.api-bridge/pull/25 encapsulates functionality... closing

**Pull request closed without merging**